File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    max = 0
    def maxDepth(self, root, depth=0):
        if root == None :
            return 0
        if root.left == None and root.right == None :
            return depth + 1
        left = self.maxDepth(root.left,depth+1)
        right = self.maxDepth(root.right,depth+1)
    
        return max(left,right)
    
        

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.kt
/**
 * Example:
 * var ti = TreeNode(5)
 * var v = ti.`val`
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun dfs(root:TreeNode?) : Int{
        if (root == null){
            return 0
        }
        val left = dfs(root.left)
        val right = dfs(root.right)
        return maxOf(left,right)+1
    }
    fun maxDepth(root: TreeNode?): Int {
        return dfs(root)
    }
}